driver-core/src/main/com/mongodb/internal/connection/FutureAsyncCompletionHandler.java

```diff
@@ -22,56 +22,41 @@
 import com.mongodb.lang.Nullable;
 
 import java.io.IOException;
-import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import static com.mongodb.internal.thread.InterruptionUtil.interruptAndCreateMongoInterruptedException;
 
-class FutureAsyncCompletionHandler<T> implements AsyncCompletionHandler<T> {
```

Comment:
Somehow this class exists but is unused, and there are two nearly identical inner classes.